app/notification/schedular/task_executer_service.py

```diff
@@ -0,0 +1,124 @@
+            self.logger.info(f"{thread_id} Processed the message: {massage_id}")
+            return message
+        except Exception as e:
+            self.logger.error(f"An error occurred while processing the message {massage_id}", e)
+
+    def _handle_message_receiving_and_processing(self):
+        """
+        Handle message retrieve from the SQS service and get the json from S3 bucket
+        """
+        current_thread = threading.current_thread()
+        thread_id = f"[{current_thread.name}:{current_thread.ident}]"
+        running_threads = []
+        read_msg_ids = []
+        if Config.TASK_EXECUTOR_MAX_THREAD >= self.executor.queue_size():
+            batch_size = Config.SQS_BATCH_SIZE
+            visibility_time = Config.SQS_VISIBILITY_TIME
+            wait_time = Config.SQS_WAIT_TIME
+            notification_messages = self.sqs_extended_client.receive_messages(
+                Config.AWS_SQS_NOTIF_APP_PRIMARY_QUEUE_URL,
+                batch_size,
+                visibility_time,
+                wait_time,
+            )
+            self.logger.debug(f"{thread_id} Message Count [{len(notification_messages)}]")
+            if notification_messages:
+                for message in notification_messages:
+                    message_id = message["sqs"]["MessageId"]
+                    self.logger.info(f"{thread_id} Message id [{message_id}]")
+                    read_msg_ids.append(message["sqs"]["MessageId"])
+                    task = self.executor.submit(self._notification_task, message)
+                    running_threads.append(task)
+        else:
+            self.logger.info(f"{thread_id} Max thread limit reached hence stop reading messages from queue")
+
+        self.logger.debug(
+            f"{thread_id} Received Message count [{len(read_msg_ids)}] "
+            f"Created thread count [{len(running_threads)}]"
+        )
+        return running_threads, read_msg_ids
+
+    def _handle_message_delete_processing(self, running_threads, read_msg_ids):
+        """
+        Handling the message delete process from the SQS and S3 bucket if it is completed
+        :param read_msg_ids All the message ids that taken from SQS
+        :param running_threads Executing tasks to send emails
+        """
+        current_thread = threading.current_thread()
+        thread_id = f"[{current_thread.name}:{current_thread.ident}]"
+        receipt_handles_to_delete = []
+        completed_msg_ids = []
+        for future in as_completed(running_threads):
+            try:
+                msg = future.result()
+                msg_id = msg["sqs"]["MessageId"]
+                receipt_handles_to_delete.append(msg["sqs"])
+                completed_msg_ids.append(msg_id)
+                self.logger.debug(f"{thread_id} Execution completed and deleted from queue: {msg_id}")
+            except Exception as e:
+                self.logger.error(f"{thread_id} An error occurred while processing the message {e}")
```

Comment:
If the message wasn't processed do we need to move it to the dead letter queue? Or is that done somewhere else?

Reply:
Dead letter queue we don't need to handle if we did not send the delete it will be automatically added to dead letter from AWS side it should handle